xline/src/server/lease_server.rs

```diff
@@ -312,13 +258,28 @@ where
         request: tonic::Request<tonic::Streaming<LeaseKeepAliveRequest>>,
     ) -> Result<tonic::Response<Self::LeaseKeepAliveStream>, tonic::Status> {
         debug!("Receive LeaseKeepAliveRequest {:?}", request);
-        let request_stream = request.into_inner();
-        let response_stream = if self.is_leader() {
-            self.leader_keep_alive(request_stream).await
+        let mut request_stream = request.into_inner();
+        if self.is_leader() {
+            let stream = self.leader_keep_alive(request_stream).await;
+            Ok(tonic::Response::new(stream))
         } else {
-            self.follower_keep_alive(request_stream).await?
-        };
-        Ok(tonic::Response::new(response_stream))
+            let leader_addr = self.state.wait_leader().await?;
```

Comment:
Shall we put the logic to the `follower_keep_alive` function?

Reply:
Fixed